File: 卡码/代码随想录/1/06-二叉树/Go/01-二叉树遍历方式/LeetCode_0637.go
//637. 二叉树的层平均值
package main

import (
	. "binaryTree/types"
)

func averageOfLevels(root *TreeNode) []float64 {
	res:=make([]float64,0)
	q:=make([]*TreeNode,0)
	var size,sum int
	if root!=nil {
		q=append(q,root)
	}
	for len(q)>0 {
		size=len(q)
		sum=0
		for i:=0;i<size;i++ {
			node:=q[0]
			q=q[1:]
			sum+=node.Val
			if node.Left!=nil {
				q=append(q,node.Left)
			}
			if node.Right!=nil {
				q=append(q,node.Right)
			}
		}
		res=append(res,float64(sum)/float64(size))
	}
	return res
}

File: 卡码/代码随想录/1/06-二叉树/Js/01-二叉树遍历方式/LeetCode_0637.js
//637. 二叉树的层平均值
import TreeNode from "../TreeNode.js";

const averageOfLevels = function(root) {
  const res=new Array();
  const q=new Array();
  let size=0,sum=0;
  if(root) q.push(root);
  while(q.length>0){
    size=q.length;
    sum=0;
    for(let i=0;i<size;i++){
      let node=q.shift();
      sum+=node.val;
      if(node.left) q.push(node.left);
      if(node.right) q.push(node.right);
    }
    res.push(sum/size);
  }
  return res;
};